ansible/roles/rt/tasks/main.yml

```diff
@@ -1,38 +1,147 @@
-#    set_fact: db_host={{docker_containers[0].NetworkSettings.IPAddress}}
-#
-#  - name: show db_host
-#    debug: msg="{{db_host}}"
-
-- name: check if our rt container is there
-  shell: docker ps | tail -n +2 | awk '{ print $NF }' | sed 's/,/\n/g' | grep '^rt$'
-  register: rt
-  ignore_errors: True
-
-- name: docker rt volumes
-  command: mkdir -p /data/{{ docker_container }}
-  when: rt.rc == 1
-
-- name: create docker rt server
-  command: docker run -d -e="RT_DATABASE_PW={{rt_database_pw}}" -v="/data/rt:/data" -p=127.0.0.1:5025:25 -p=127.0.0.1:5080:80 -link=rt_pg:db -name={{ docker_container }} okfn/rt:0.16
-  when: rt.rc == 1
-
-- name: configure docker rt server
-  copy: src=RT_SiteConfig.pm dest=/data/rt/rt_conf/ owner=root group=www-data mode=0640
-  notify: restart rt
+
+- name: Install RT pre-reqs through apt-get
+  apt: pkg={{ item }} state=installed
+  with_items:
+  - build-essential 
+  - postgresql-client 
+  - apache2 
+  - mailutils 
+  - openssl 
+  - libyaml-perl 
+  - libyaml-appconfig-perl 
+  - libssl-dev 
+  - libexpat1-dev
+  - libdbd-mysql-perl
+  - libnet-smtp-ssl-perl 
+  - libdatetime-event-ical-perl
+  - procmail
+
+- name: get latest RT release
+  get_url: url=https://download.bestpractical.com/pub/rt/release/rt-4.2.9.tar.gz dest=/opt/
+
+- name: extract RT 
+  unarchive: src=/opt/rt-4.2.9.tar.gz dest=/opt/ copy=no
+
+- name: copy rt dir to rt4 dir
+  command: mv /opt/rt-4.2.9 /opt/rt4-install
+
+- name: configure RT install
+  command: ./configure
+  args:  
+    chdir: /opt/rt4-install
+
+- name: configure cpan
+  command: perl -MCPAN -e 'my $c = "CPAN::HandleConfig"; $c->load(doit => 1, autoconfig => 1); $c->edit(prerequisites_policy => "follow"); $c->edit(build_requires_install_policy => "yes"); $c->commit'
+  args: 
+    chdir: /opt/rt4-install
+  async: 3600
+  poll: 60
+
+- name: install RTx to install RT extensions
+  command: /usr/bin/perl -MCPAN -e 'install Module::Install::RTx::Runtime'
+
+- name: make fixdeps
+  shell: cd /opt/rt4-install && make fixdeps
+  async: 3600
+  poll: 60
+
+- name: make install
+  command: make install
+  args:
+    chdir: /opt/rt4-install
+
+# Extra plugins
+# To add an extra plugin, just add it to the with_items below
+# Make sure you copy the format in the with items
+# It assumes you pick-up the files from github
+# Make sure you check the play after each plugin is added
+
+- name: make directories to put the plugins in
+  file: path=/opt/{{ item }} state=directory
+  with_items:
+    - RT-Extension-ActivityReports
+    - rt-extension-commandbymail
+    - rt-extension-nagios
+    - rt-extension-mergeusers
+    - rt-extension-repeatticket
+    - rt-extension-resetpassword
+    - rt-extension-spawnlinkedticketinqueue
+
+- name: get nagios all nagios plugins
+  get_url: url={{ item.url }} dest=/opt/{{ item.name }}/master.zip
+  with_items:
+    - { url: 'https://github.com/bestpractical/rt-extension-activityreports/archive/master.zip', name: 'RT-Extension-ActivityReports' }
```

Comment:
Pulling master for everything is problematic. There may be something in the future that'll break with our version of RT. Is there a way to fix a commit? Or perhaps git clone it to a particular commit?